auth-jwt-service/src/test/java/io/stargate/auth/jwt/AuthJwtServiceTest.java

```diff
@@ -0,0 +1,108 @@
+package io.stargate.auth.jwt;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTClaimsSet.Builder;
+import io.stargate.auth.StoredCredentials;
+import io.stargate.auth.UnauthorizedException;
+import java.net.MalformedURLException;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+public class AuthJwtServiceTest {
+
+  private static final JwtValidator jwtValidator =
+      mock(JwtValidator.class, withSettings().useConstructor("http://example.com"));
```

Comment:
Would it be possible to mock `DefaultJWTProcessor` for deeper test coverage?

Reply:
I'm not sure what that gets us since the method is only

```
public JWTClaimsSet validate(String token) throws UnauthorizedException {
    JWTClaimsSet claimsSet;
    try {
      claimsSet = jwtProcessor.process(token, null); // context is an optional param so passing null
    } catch (ParseException | JOSEException e) {
      logger.error("Failed to process JWT", e);
      throw new RuntimeException(e);
    } catch (BadJOSEException badJOSEException) {
      logger.info("Tried to validate invalid JWT", badJOSEException);
      throw new UnauthorizedException("Invalid JWT");
    }

    return claimsSet;
  }
```

Mocking `DefaultJWTProcessor` is only going to expand coverage into the catch blocks and at that point it's roughly the same as mocking the `validate()` method and returning exceptions, right? Unless I'm missing something.

Reply:
I was mostly thinking about the `JwtValidator` constructor. In current test we invoke it, but do not validate how it handles the URL... but it's a minor point.